.github/workflows/smoke-test.yml

```
@@ -33,7 +33,7 @@ jobs:
 
   smoke-test:
     needs: build
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
```

How about adding 20.04 and 24.04? The snap should be tested on all.
```suggestion
strategy:
    matrix:
        os: [ubuntu-24.04, ubuntu-22.04, ubuntu-20.04]
    runs-on: ${{ matrix.os }}
```